src/reviewer.py

```diff
@@ -0,0 +1,46 @@
+import os
+from github_utils import get_pull_request_diff, post_review_comment
+from llm_utils import analyze_code_changes
+
+class AICodeReviewer:
+    def __init__(self):
+        self.openai_api_key = os.getenv('GEMINI_API_KEY')
+        self.github_token = os.getenv('GIT_TOKEN')
+
+    def review_pull_request(self):
+        try:
+            # Get the PR diff
+            diff_content = get_pull_request_diff()
+            if not diff_content:
+                print("No diff content found")
+                return
+
+            # Analyze changes using LLM
+            review_comments = analyze_code_changes(diff_content)
+
+            if not review_comments:
+                print("No comments generated by LLM")
+                return
+
+
+            print("Generated comments preview:")
+            for idx, comment in enumerate(review_comments[:5], 1):
+                print(f"Comment {idx}:")
+                print(f"File: {comment.get('path', '')}")
+                print(f"Line: {comment.get('line', '')}")
```

Comment:
Consider using logging instead of print statements for better debugging and monitoring.

```suggestion
|
```
 import logging


 logging.basicConfig(level=logging.INFO)


 ...
 logging.info("No diff content found")
```